### PR TITLE
test: e2e test chash upstream with key (query_string, arg_xxx)

### PR DESCRIPTION
**Pull request by idbeta**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] E2E test
- [ ] New feature provided
- [ ] Improve performance

related issue: https://github.com/apache/apisix-dashboard/issues/908

### CODE REVIEW

File: api/test/e2e/upstream_chash_query_string_arg_xxx_test.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+		respBody, err = ioutil.ReadAll(resp.Body)
+		body := string(respBody)
+		if _, ok := res[body]; !ok {
+			res[body] = 1
+		} else {
+			res[body] += 1
+		}
+		resp.Body.Close()
+	}
+	//the results will be distributed among the 3 upstream
```
Comment (membphis) [inline api/test/e2e/upstream_chash_query_string_arg_xxx_test.go:97]: please add a comment: `todo: need to find a better way to confirm the result`

Comment (membphis) [reply api/test/e2e/upstream_chash_query_string_arg_xxx_test.go:97]: current way is unstable

File: api/test/e2e/upstream_chash_query_string_arg_xxx_test.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
+	}
+
+	for _, tc := range tests {
+		testCaseCheck(tc)
+	}
+
+	//hit routes
```
Comment (membphis) [inline api/test/e2e/upstream_chash_query_string_arg_xxx_test.go:77]: bad comment style, please fix other similar points

File: api/test/e2e/upstream_chash_query_string_arg_xxx_test.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/1",
+			Body: `{
+				"uri": "/server_port",
+				"upstream_id": "1"
+			}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
```
Comment (membphis) [inline api/test/e2e/upstream_chash_query_string_arg_xxx_test.go:144]: https://github.com/apache/apisix-dashboard/pull/932/files#diff-13ef2604ee721f34005fc15685defc18abb4562ea1a2a95d7931e89b0f59c19dR132-R144

we can delete this step.

Comment (idbeta) [reply api/test/e2e/upstream_chash_query_string_arg_xxx_test.go]: fixed.

**Review by idbeta — COMMENTED** [review]
fixed.

File: api/test/e2e/upstream_chash_query_string_arg_xxx_test.go (hunk 2 above)
Comment (idbeta) [reply api/test/e2e/upstream_chash_query_string_arg_xxx_test.go]: fixed.

File: api/test/e2e/upstream_chash_query_string_arg_xxx_test.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+			res[body] += 1
+		}
+		resp.Body.Close()
+	}
+	var counts []int
+	for _, value := range res {
+		counts = append(counts, value)
+	}
+	sort.Ints(counts)
+	assert.True(t, float64(counts[2]-counts[0]) / float64(counts[1]) < 0.4) // the result is unstable, fix it later
```
Comment (membphis) [inline api/test/e2e/upstream_chash_query_string_arg_xxx_test.go:103]: add a `todo` keyword

File: api/test/e2e/upstream_chash_query_string_arg_xxx_test.go [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
```
Comment (membphis) [inline api/test/e2e/upstream_chash_query_string_arg_xxx_test.go:166]: ditto

**Review by membphis — APPROVED** [review]
LGTM, except minor things

**Comment by nic-chen** [comment]
ping @idbeta

File: api/test/e2e/upstream_chash_query_string_arg_xxx_test.go (hunk 4 above)
Comment (idbeta) [reply api/test/e2e/upstream_chash_query_string_arg_xxx_test.go]: fixed.

File: api/test/e2e/upstream_chash_query_string_arg_xxx_test.go (hunk 3 above)
Comment (idbeta) [reply api/test/e2e/upstream_chash_query_string_arg_xxx_test.go]: fixed

File: api/test/e2e/upstream_chash_query_string_arg_xxx_test.go (hunk 1 above)
Comment (idbeta) [reply api/test/e2e/upstream_chash_query_string_arg_xxx_test.go]: fixed.

**Review by nic-chen — APPROVED** [review]
LGTM Except for some code format issues

**Pull request merged**